src/app/trainees/trainee-search/trainee-search.component.ts

```diff
@@ -0,0 +1,58 @@
+import { Component, OnInit } from "@angular/core";
+import { FormBuilder, FormGroup, Validators } from "@angular/forms";
+import { ActivatedRoute, Params } from "@angular/router";
+import { Select, Store } from "@ngxs/store";
+import { Observable } from "rxjs";
+import { take } from "rxjs/operators";
+import {
+  GetTrainees,
+  SearchTrainees,
+  UpdateTraineesRoute
+} from "../state/trainees.actions";
+import { TraineesState } from "../state/trainees.state";
+
+@Component({
+  selector: "app-trainee-search",
+  templateUrl: "./trainee-search.component.html"
+})
+export class TraineeSearchComponent implements OnInit {
+  @Select(TraineesState.searchQuery) searchQuery$: Observable<string>;
+  public form: FormGroup;
+  public params: Params = this.route.snapshot.queryParams;
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private store: Store,
+    private route: ActivatedRoute
+  ) {}
+
+  ngOnInit() {
+    this.setupForm();
+  }
+
+  public setupForm(): void {
+    this.form = this.formBuilder.group({
+      searchQuery: [this.params.searchQuery, [Validators.minLength(3)]]
```

Comment:
discuss validator if its really needed.

Reply:
ify has updated the jira ticket so minLength is now set at 2 characters. pr updated